changedetectionio/__init__.py

```diff
@@ -597,13 +620,145 @@ def import_page():
+                            update_q.put(uuid)
+                            i += 1
+                    
+                except KeyError :
+                    pass
+
+                flash("Rechecking {0} watch{1}.".format(i, "" if i == 1 else "es"))
+            
+            # Clear selected statuses, so we do not see the 'unviewed' class
+            elif func == 'mark_selected_viewed' :
+
+                try :
+                    for uuid in uuids.split(',') :
+                        datastore.data['watching'][uuid]['last_viewed'] = datastore.data['watching'][uuid]['newest_history_key']
+                
+                    # set viewed to false
+                    datastore.data['watching'][uuid]['viewed'] = False
+               
+                except KeyError :
+                    pass
+
+                datastore.needs_write = True
+
+            # Reset selected statuses, so we see the 'unviewed' class
+            # both funcs will contain the uuid list from the processChecked javascript function
+            elif func == 'mark_selected_notviewed' or func == 'mark_all_notviewed' :
+
+                # count within limit_tag and count successes and capture unchanged
+                tagged = 0
+                marked = 0
+                unchanged = []
+                
+                try :
+                    for uuid in uuids.split(',') :
+                        # increment count with limit_tag
+                        tagged += 1
+                        dates = list(datastore.data['watching'][uuid]['history'].keys())
+                        # Convert to int, sort and back to str again
+                        dates = [int(i) for i in dates]
+                        dates.sort(reverse=True)
+                        dates = [str(i) for i in dates]
+
+                        # must be more than 1 history to mark as not viewed
+                        if len(dates) > 1 :
+                            # Save the next earliest history as the most recently viewed
+                            datastore.set_last_viewed(uuid, dates[1])
+                            # set viewed to false
+                            datastore.data['watching'][uuid]['viewed'] = False
+                            # increment successes
+                            marked += 1
+                            
+                        else : 
+                            if datastore.data['watching'][uuid]['title'] :
+                                unchanged.append(datastore.data['watching'][uuid]['title'])
+                            else :
+                                unchanged.append(datastore.data['watching'][uuid]['url'])
+
+                except KeyError :
+                        pass
+
+                datastore.needs_write = True
+
+                if marked < tagged :
+                    flash("The following {} not have enough history to be remarked.".format("watch does" if len(unchanged) == 1 else "watches do"), "notice")
```

Comment:
grammatical here maybe "do not have.."

Reply:
seems ok, but swapped the period out for a colon, lol.

![image](https://user-images.githubusercontent.com/16741693/145522110-950accca-8cd8-4b67-bf3b-0b78c5d5356f.png)

![image](https://user-images.githubusercontent.com/16741693/145522153-c7e3671a-1e64-4fec-b2fd-46ac48de113e.png)